pkg/performanceprofile/controller/performanceprofile/resources/resources.go

```diff
@@ -27,20 +28,20 @@ func mergeMaps(src map[string]string, dst map[string]string) {
 
 // TODO: we should merge all create, get and delete methods
 
-func (r *PerformanceProfileReconciler) getMachineConfig(ctx context.Context, name string) (*mcov1.MachineConfig, error) {
+func GetMachineConfig(ctx context.Context, client client.Client, name string) (*mcov1.MachineConfig, error) {
```

Comment:
is there a way to use the client without passing it as an argument in every function ? (same for all functions below)

Reply:
We need to pass the client from outside because these function will be used for both hypershift and non-hypershift clusters. 
Each cluster will be implementing its own logic, hence it should control the type of client it wants to pass.